src/kalshi.rs

```diff
@@ -0,0 +1,377 @@
+            liquidity: question.get_market().liquidity,
+            threshold: requirements.min_liquidity,
+        });
+    }
+    // Min volume
+    if question.get_market().volume < requirements.min_volume {
+        return Err(KalshiCheckFailure::NotEnoughVolume {
+            volume: question.get_market().volume,
+            threshold: requirements.min_volume,
+        });
+    }
+    // Min recent volume
+    if question.get_market().recent_volume < requirements.min_recent_volume {
+        return Err(KalshiCheckFailure::NotEnoughRecentVolume {
+            recent_volume: question.get_market().recent_volume,
+            threshold: requirements.min_recent_volume,
+        });
+    }
+    // Min open interest
+    if question.get_market().open_interest < requirements.min_open_interest {
+        return Err(KalshiCheckFailure::NotEnoughOpenInterest {
+            open_interest: question.get_market().open_interest,
+            threshold: requirements.min_open_interest,
+        });
+    }
+    // min dollar volume
+    if question.get_market().dollar_volume < requirements.min_dollar_volume {
+        return Err(KalshiCheckFailure::NotEnoughDollarVolume {
+            dollar_volume: question.get_market().dollar_volume,
+            threshold: requirements.min_dollar_volume,
+        });
+    }
+    // min dollar recent volume
+    if question.get_market().dollar_recent_volume < requirements.min_dollar_recent_volume {
+        return Err(KalshiCheckFailure::NotEnoughDollarRecentVolume {
+            dollar_recent_volume: question.get_market().dollar_recent_volume,
+            threshold: requirements.min_dollar_recent_volume,
+        });
+    }
+    // min dollar open interest
+    if question.get_market().dollar_open_interest < requirements.min_dollar_open_interest {
+        return Err(KalshiCheckFailure::NotEnoughDollarOpenInterest {
+            dollar_open_interest: question.get_market().dollar_open_interest,
+            threshold: requirements.min_dollar_open_interest,
+        });
+    }
+
+    if question.time_to_resolution() < Duration::days(requirements.min_days_to_resolution) {
+        return Err(KalshiCheckFailure::ResolvesTooSoon {
+            days_remaining: question.time_to_resolution().num_days(),
+            threshold: requirements.min_days_to_resolution,
+        });
+    }
+    if question.time_to_resolution() > Duration::days(requirements.max_days_to_resolution) {
+        return Err(KalshiCheckFailure::ResolvesTooLate {
+            days_remaining: question.time_to_resolution().num_days(),
+            threshold: requirements.max_days_to_resolution,
+        });
+    }
+    if question.age() > Duration::days(requirements.max_age_days) {
+        return Err(KalshiCheckFailure::TooOld {
+            age_days: question.age().num_days(),
+            threshold: requirements.max_age_days,
+        });
+    }
+    if (100 - question.get_market().yes_ask) > requirements.max_confidence * 100
```

Comment:
Kind of puzzled this compiles, since max_confidence is presumably a float.

Reply:
Yeah, looks like it converts it to 0. Not compiling would have been great.